plugins/action/generate_spec.py

```diff
@@ -0,0 +1,92 @@
+# (c) 2020, Ansible by Red Hat, inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible.plugins.action import ActionBase
+import os
+from ansible.errors import AnsibleError
+
+from ansible.module_utils._text import to_text
+YANG_SPEC_DIR_PATH = "~/.ansible/tmp/yang_spec"
+from ansible_collections.community.yang.plugins.lookup.spec import (
```

Comment:
The spec generation logic should be reused from module_utils code and not directly refer to LookupModule code.

Reply:
Incorporated